app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtTextActions.java

```diff
@@ -164,7 +170,7 @@ public void onClick(View view) {
                     return;
                 }
                 case R.string.tmaid_todotxt_current_date: {
-                    _hlEditor.getText().insert(origSelectionStart, SttCommander.getToday());
+                    setKeyDate(origTask, "due", 3, getContext().getString(R.string.set_due_date));
```

Comment:
Note that if multiple lines are selected, the first one will be used to get the initial due date, but all will be updated.